src/Microsoft.AspNetCore.Mvc.Analyzers/ApiConventionAnalyzer.cs

```diff
@@ -0,0 +1,303 @@
+                if (statusCode == null)
+                {
+                    // Unable to read the status code. Treat this as valid.
+                    return null;
+                }
+
+                context.ActualResponseMetadata.Add(statusCode.Value);
+                if (!HasStatusCode(context.ExpectedResponseMetadata, statusCode.Value))
+                {
+                    return Diagnostic.Create(
+                        DiagnosticDescriptors.MVC1004_ActionReturnsUndocumentedStatusCode,
+                        location,
+                        statusCode);
+                }
+            }
+            else if (!context.SymbolCache.IActionResult.IsAssignableFrom(context.MethodReturnType) &&
+                context.MethodReturnType.IsAssignableFrom(type))
+            {
+                if (!HasStatusCode(context.ExpectedResponseMetadata, 200) && !HasStatusCode(context.ExpectedResponseMetadata, 201))
+                {
+                    return Diagnostic.Create(
+                        DiagnosticDescriptors.MVC1005_ActionReturnsUndocumentedSuccessResult,
+                        location);
+                }
+            }
+
+            return null;
+        }
+
+        internal static int? GetDefaultStatusCode(AttributeData attribute)
+        {
+            if (attribute != null &&
+                attribute.ConstructorArguments.Length == 1 &&
+                attribute.ConstructorArguments[0].Kind == TypedConstantKind.Primitive &&
+                attribute.ConstructorArguments[0].Value is int statusCode)
+            {
+                return statusCode;
+            }
+
+            return null;
+        }
+
+        internal static ITypeSymbol UnwrapMethodReturnType(
+            in ApiControllerSymbolCache symbolCache,
+            SemanticModel semanticModel,
+            IMethodSymbol method)
+        {
+            var location = method.Locations.Length > 0 ? method.Locations[0] : Location.None;
+
+            var returnType = method.ReturnType;
+            var awaitableReturnType = returnType.InferAwaitableReturnType(semanticModel, location.SourceSpan.Start);
+            returnType = awaitableReturnType ?? returnType;
+
+            if (returnType is INamedTypeSymbol namedReturnType &&
+                namedReturnType.IsGenericType &&
+                namedReturnType.TypeArguments.Length == 1 &&
+                namedReturnType.ConstructedFrom != null &&
+                symbolCache.ActionResultOfT.IsAssignableFrom(namedReturnType.ConstructedFrom))
+            {
+                returnType = namedReturnType.TypeArguments[0];
+            }
+
+            return returnType;
+        }
+
+        private static bool ShouldEvaluateMethod(ApiControllerSymbolCache symbolCache, IMethodSymbol method)
+        {
+            if (method == null)
+            {
+                return false;
+            }
+
+            if (method.ReturnsVoid || method.ReturnType.TypeKind == TypeKind.Error)
```

Comment:
Probably worth it to ignore the entire method if it has any contained errors as well. Basically, if there's anything inside of it that results in an error. Ignore it.

In other locations you error check as well is why I bring it up.

Reply:
There isn't a way to do that - method symbols are always resolved (although they may have invalid syntax which is what the `IsMissing` thing is about), but you could be missing types.